dateparser/conf.py

```diff
@@ -79,8 +81,133 @@ def wrapper(*args, **kwargs):
             kwargs['settings'] = settings.replace(mod_settings=mod_settings, **kwargs['settings'])
 
         if not isinstance(kwargs['settings'], Settings):
-            raise TypeError(
-                "settings can only be either dict or instance of Settings class")
+            raise TypeError("settings can only be either dict or instance of Settings class")
 
         return f(*args, **kwargs)
     return wrapper
+
+
+class SettingValidationError(ValueError):
+    pass
+
+
+def _check_require_part(provided_list):
+    """Returns `True` if the provided list of parts is valid"""
+    return not set(provided_list) - {'day', 'month', 'year'}
+
+
+def _check_parsers(provided_list):
+    """Returns `True` if the provided list of parsers is valid"""
+    existing_parsers = ['timestamp', 'relative-time', 'custom-formats', 'absolute-time', 'no-spaces-time']
```

Comment:
I feel like `existing_parsers` should be imported from somewhere else. And I think it would be useful for users to be able to import it as well, e.g. to write something like:

```python
undesired_parsers = {'foo', 'bar'}
parsers_to_use = [parser for parser in EXISTING_PARSER if parser not in undesired_parsers]
```

Reply:
I think this too: https://github.com/scrapinghub/dateparser/issues/798

However, I don't want to leave this without merging after fixing that issue, so I prefer to include it as-is and change it once we fix #798